### ISSUE TITLE
solr doc manager fails for _ids with ":" (colons)

### ISSUE THREAD

**Issue by bobend** [10gen-labs/mongo-connector#209]
If you try to sync a collection with [URN](http://en.wikipedia.org/wiki/Uniform_resource_name)s as _ids the solr doc manager in mongo-connector seems to send invalid data for updates. 
`org.apache.lucene.queryParser.ParseException: Cannot parse '_id:urn:dr:mu:programcard:4f3b88e2860d9a33ccfdada8': Encountered " ":" ": "" at line 1, column 7.`
The initial dump works ok, but after that is done updates fails.
Are the ids escaped incorrectly?

I running with python 2.6.6 and solr 3.6.2

This used to work in an older version of the solr doc manager (  714d065ed0e4e6bcd4ab92ff7baa383dac3f9e93  )

example document
```json 
{
  "_id" : "urn:dr:mu:programcard:4f3b88e2860d9a33ccfdada8",
  "Title" : "Molly Monster",
  "Description" : "Man kan forestille sig de særeste ting."
} 
 ```
Error from mongo-connector.log:
```

[ERROR] mongo_connector.oplog_manager:278 - Unable to process oplog document ... 


Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/mongo_connector-2.0.1-py2.6.egg/mongo_connector/oplog_manager.py", line 264, in run
    ns, timestamp)
  File "/usr/lib/python2.6/site-packages/mongo_connector-2.0.1-py2.6.egg/mongo_connector/util.py", line 38, in wrapped
    reraise(new_type, exc_value, exc_tb)
  File "/usr/lib/python2.6/site-packages/mongo_connector-2.0.1-py2.6.egg/mongo_connector/util.py", line 32, in wrapped
    return f(*args, **kwargs)
  File "/usr/lib/python2.6/site-packages/mongo_connector-2.0.1-py2.6.egg/mongo_connector/doc_managers/solr_doc_manager.py", line 230, in update
    results = self.solr.search(query)
  File "/usr/lib/python2.6/site-packages/pysolr.py", line 616, in search
    response = self._select(params)
  File "/usr/lib/python2.6/site-packages/pysolr.py", line 334, in _select
    return self._send_request('get', path)
  File "/usr/lib/python2.6/site-packages/pysolr.py", line 322, in _send_request
    raise SolrError(error_message)
OperationFailed: [Reason: null]
```
Solr error from `/var/log/tomcat6/catalina.out`:
```
SEVERE: org.apache.solr.common.SolrException
        at org.apache.solr.handler.component.QueryComponent.prepare(QueryComponent.java:134)
        at org.apache.solr.handler.component.SearchHandler.handleRequestBody(SearchHandler.java:165)
        at org.apache.solr.handler.RequestHandlerBase.handleRequest(RequestHandlerBase.java:129)
        at org.apache.solr.core.SolrCore.execute(SolrCore.java:1376)
        at org.apache.solr.servlet.SolrDispatchFilter.execute(SolrDispatchFilter.java:365)
        at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:260)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:235)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:233)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:191)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:102)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:109)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:298)
        at org.apache.coyote.http11.Http11Processor.process(Http11Processor.java:857)
        at org.apache.coyote.http11.Http11Protocol$Http11ConnectionHandler.process(Http11Protocol.java:588)
        at org.apache.tomcat.util.net.JIoEndpoint$Worker.run(JIoEndpoint.java:489)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.apache.lucene.queryParser.ParseException: Cannot parse '_id:urn:dr:mu:programcard:4f3b88e2860d9a33ccfdada8': Encountered " ":" ": "" at line 1, column 7.```

**Comment by lovett89** [comment]
Turns out the document id is not escaped at all. I'll fix this now.

**Comment by lovett89** [comment]
Thanks again for the report. This is fixed in 5f9c8a062e3b5181762b9cf9ec280d8985d5fe07.
